larray/core.py

```diff
@@ -7661,8 +7661,11 @@ def percentile(self, q, *args, **kwargs):
         if skipna is None:
             skipna = True
         func = np.nanpercentile if skipna else np.percentile
-        return self._aggregate(func, args, kwargs, keepaxes=keepaxes,
-                               commutative=True, extra_kwargs={'q': q})
+        if isinstance(q, (list, tuple)):
+            return stack([(v, self._aggregate(func, args, kwargs, keepaxes=keepaxes, commutative=True,
+                                              extra_kwargs={'q': v})) for v in q], 'percentile')
```

Comment:
OK in case of axes but in _group_aggregate(), op(arr, axis=axis_idx, out=out, **kwargs) may be called several times and creates a new dimension each time.